### ISSUE TITLE
Issue with thread posts on 5.6.0

### ISSUE THREAD

**Issue by PsychoSid** [death-claw/vripper-project#184]
Just updated and trying to pull multiple posts from a thread and get. Thanks for all the work. 

```
Caused by: org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PUBLIC.POST__POST_ID_UQ_INDEX_2 ON PUBLIC.POST(POST_ID NULLS FIRST) VALUES ( /* 1664 */ CAST(173402515 AS BIGINT) )"; SQL statement:
INSERT INTO POST (ADDED_AT, DONE, DOWNLOADED, FOLDER_NAME, FORUM, HOSTS, OUTPUT_PATH, POST_ID, POST_TITLE, "RANK", "SIZE", STATUS, THREAD_ID, THREAD_TITLE, TOKEN, TOTAL, URL) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-224]
        at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1282)
        at org.jetbrains.exposed.sql.statements.jdbc.JdbcPreparedStatementImpl.executeBatch(JdbcPreparedStatementImpl.kt:91)
        at org.jetbrains.exposed.sql.statements.InsertStatement.execInsertFunction(InsertStatement.kt:158)
        at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:168)
        at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:18)
        at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:92)
        ... 35 more
org.jetbrains.exposed.exceptions.ExposedSQLException: org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PUBLIC.POST__POST_ID_UQ_INDEX_2 ON PUBLIC.POST(POST_ID NULLS FIRST) VALUES ( /* 1664 */ CAST(173402515 AS BIGINT) )"; SQL statement:
INSERT INTO POST (ADDED_AT, DONE, DOWNLOADED, FOLDER_NAME, FORUM, HOSTS, OUTPUT_PATH, POST_ID, POST_TITLE, "RANK", "SIZE", STATUS, THREAD_ID, THREAD_TITLE, TOKEN, TOTAL, URL) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-224]
SQL: [Failed on expanding args for INSERT: org.jetbrains.exposed.sql.statements.BatchInsertStatement@7abe0db9]
        at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:94)
        at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:248)
        at org.jetbrains.exposed.sql.Transaction.exec(Transaction.kt:225)
        at org.jetbrains.exposed.sql.statements.Statement.execute(Statement.kt:55)
        at org.jetbrains.exposed.sql.QueriesKt.executeBatch(Queries.kt:302)
        at org.jetbrains.exposed.sql.QueriesKt.batchInsert(Queries.kt:201)
        at org.jetbrains.exposed.sql.QueriesKt.batchInsert(Queries.kt:176)
        at org.jetbrains.exposed.sql.QueriesKt.batchInsert$default(Queries.kt:171)
        at me.vripper.repositories.impl.PostDownloadStateRepositoryImpl.save(PostDownloadStateRepositoryImpl.kt:19)
        at me.vripper.services.DataTransaction$save$1.invoke(DataTransaction.kt:32)
        at me.vripper.services.DataTransaction$save$1.invoke(DataTransaction.kt:32)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$transaction$1.invoke(ThreadLocalTransactionManager.kt:239)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:361)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:231)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:209)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:208)
        at me.vripper.services.DataTransaction.save(DataTransaction.kt:32)
        at me.vripper.services.DataTransaction.access$save(DataTransaction.kt:19)
        at me.vripper.services.DataTransaction$newPosts$savedPosts$1.invoke(DataTransaction.kt:122)
        at me.vripper.services.DataTransaction$newPosts$savedPosts$1.invoke(DataTransaction.kt:121)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:306)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.access$inTopLevelTransaction$run(ThreadLocalTransactionManager.kt:1)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$inTopLevelTransaction$1.invoke(ThreadLocalTransactionManager.kt:353)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:361)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.inTopLevelTransaction(ThreadLocalTransactionManager.kt:352)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt$transaction$1.invoke(ThreadLocalTransactionManager.kt:261)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.keepAndRestoreTransactionRefAfterRun(ThreadLocalTransactionManager.kt:361)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:231)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction(ThreadLocalTransactionManager.kt:209)
        at org.jetbrains.exposed.sql.transactions.ThreadLocalTransactionManagerKt.transaction$default(ThreadLocalTransactionManager.kt:208)
        at me.vripper.services.DataTransaction.newPosts(DataTransaction.kt:121)
        at me.vripper.tasks.AddPostRunnable.run(AddPostRunnable.kt:84)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: org.h2.jdbc.JdbcBatchUpdateException: Unique index or primary key violation: "PUBLIC.POST__POST_ID_UQ_INDEX_2 ON PUBLIC.POST(POST_ID NULLS FIRST) VALUES ( /* 1664 */ CAST(173402515 AS BIGINT) )"; SQL statement:
INSERT INTO POST (ADDED_AT, DONE, DOWNLOADED, FOLDER_NAME, FORUM, HOSTS, OUTPUT_PATH, POST_ID, POST_TITLE, "RANK", "SIZE", STATUS, THREAD_ID, THREAD_TITLE, TOKEN, TOTAL, URL) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-224]
        at org.h2.jdbc.JdbcPreparedStatement.executeBatch(JdbcPreparedStatement.java:1282)
        at org.jetbrains.exposed.sql.statements.jdbc.JdbcPreparedStatementImpl.executeBatch(JdbcPreparedStatementImpl.kt:91)
        at org.jetbrains.exposed.sql.statements.InsertStatement.execInsertFunction(InsertStatement.kt:158)
        at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:168)
        at org.jetbrains.exposed.sql.statements.InsertStatement.executeInternal(InsertStatement.kt:18)
        at org.jetbrains.exposed.sql.statements.Statement.executeIn$exposed_core(Statement.kt:92)
```

**Comment by death-claw** [comment]
You are not allowed to put duplicate posts

**Comment by albyofdoom** [comment]
@death-claw Could you clarify that? So if I go to a thread (`https://vipergirls.to/threads/7258399-OnlyFans-PHOTO-collection-full-sets`), and find one post that I want to download, import it and start downloading, does that mean I can't subsequently add any additional posts from that thread?

Because that's what I'm experiencing, if I add `https://vipergirls.to/threads/7258399?p=111624528&viewfull=1#post111624528`, it works fine, but if I then try to add `https://vipergirls.to/threads/7258399-OnlyFans-PHOTO-collection-full-sets?p=111624953&viewfull=1#post111624953` from the same thread, it does pauses for a sec, does nothing, and then requeues every failed or stopped post in my download queue.

So I go from ~10000 errors and 0 downloading or pending, to ~10000 pending and downloads start up. 

Prior to 5.7.0, it would do the requeuing (annoying, but harmless), but it would still add the new post and start downloading.

**Comment by death-claw** [comment]
@albyofdoom 
You can of course add multiple posts from the same thread, my first reply was precisely about duplicate posts (same thread and same post).
The error mentioned is not critical and can be ignored.

what you are describing is indeed a bug, I found out about it as well, you will find it fixed in in 5.8.0 which i'll release shortly.

**Comment by albyofdoom** [comment]
You rock, thank you for maintaining this project so well! 🙂
